crates/key-value-redis/src/store.rs

```diff
@@ -55,90 +52,69 @@ impl StoreManager for KeyValueRedis {
 }
 
 struct RedisStore {
-    connection: Arc<Mutex<MultiplexedConnection>>,
+    connection: ConnectionManager,
     database_url: Url,
 }
 
 struct CompareAndSwap {
     key: String,
-    connection: Arc<Mutex<MultiplexedConnection>>,
+    connection: ConnectionManager,
     bucket_rep: u32,
 }
 
 #[async_trait]
 impl Store for RedisStore {
+    async fn after_open(&self) -> Result<(), Error> {
+        if let Err(_error) = self.connection.clone().ping::<()>().await {
```

Comment:
Is there a reason for doing this at store open time instead of just-in-time if a command fails?  The "just pray" thing makes me worry we have a race condition if we don't allow recovery during a command...?

Reply:
If we do it per command, then there is a round-trip cost incurred for each request. Doing it this way would at least trigger the `ConnectionManager` to kickoff an automatic reconnect before any commands are issued.

Reply:
Well I was thinking do it if a command _failed_ rather than on _every_ command, so it would only incur the ping cost if actually needed rather than on every open, and that would also handle the situation where great Zeus on high Olympus spurned the praying.  (Goats.  You need more goats.)  But I'm probably imagining situations that never happen in practice - very happy to get this into users' hands as is.

Reply:
I hear you. It seems like we'd be swapping one potential race for another or at least have to implement a command retry mechanism on top of each request to ache8ve that which seems like alot more code 🫣

If you're ok with it, I think I'd like to land this as-is and revisit if weirdness persists.

Reply:
:shipit: :shipit: :shipit: